stylus/components/forms.styl

```diff
@@ -275,6 +275,71 @@ $input-text--medium
     border-radius .125rem
     padding .485rem 1rem .64rem
 
+$checkbox
+    // To make sure that checkbox's wrapper have the height of the checkbox
+    display flex
+
+    span
+        position relative
+        display inline-block
+        height checkbox-size
+        padding-left checkbox-size * 1.4
+        line-height 1rem
+        cursor pointer
+
+        &::before
+        &::after
+            content ''
+            position absolute
+            left 0
+            top 0
+            box-sizing border-box
+            width checkbox-size
+            height checkbox-size
+            border-radius .125rem
+
+        &::before
+            transition box-shadow 350ms cubic-bezier(0, .89, .44, 1)
+            box-shadow inset 0 0 0 .125rem alpha(charcoalGrey, .25)
+            background-color white
+
+            &:hover
+                box-shadow inset 0 0 0 .125rem dodgerBlue
+
+        &::after
+            background-image embedurl('../../assets/icons/ui/check.svg')
+            background-size contain
+            transition-duration .2s
+            transition-property opacity, transform
+
+    &[aria-checked='mixed'] span::after
+            background-image embedurl('../../assets/icons/ui/dash.svg')
+            background-size contain
+
+    input[type=checkbox]
+        hide()
+
+        &:checked
+            & + span::before
+                box-shadow inset 0 0 0 checkbox-size dodgerBlue
+
+            & + span::after
+                opacity 1
+                transform scale(1)
+
+        &:not(:checked) + span::after
+            opacity 0
+            transform scale(0)
+
+    &.is-error input[type=checkbox]:not(:checked) + span
+        color pomegranate
+
```

Comment:
If I understand well, a checkbox can be in error only if it's not checked ? I think it should be possible to have it in error even if it is checked.

Reply:
Well I asked myself the question and I couldn't find any usecase when checking a checkbox could be an error. Do you think of one?
Like this it's more of a security but I can change this is if needed.

Reply:
No I don't have a precise usecase here, but I find the lack of symmetry between the two cases surprising.

Reply:
Yeah I know what you mean, it's just that having a checked checkbox in red just bothered me.
Although if checked, the prop `error` should be removed by JS, so I guess it won't happen in real life.